data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/SchemaDefinitionCacheTest.java

```diff
@@ -68,4 +77,30 @@ void shouldCreateSchemaIfMilestoneRequired() {
     assertThat(spec.forMilestone(SpecMilestone.BELLATRIX)).isNull();
     assertThat(cache.getSchemaDefinition(SpecMilestone.BELLATRIX)).isNotNull();
   }
+
+  @ParameterizedTest
+  @EnumSource(SpecMilestone.class)
+  void subsequentCallsShouldGetTheSameSchemaObject(final SpecMilestone specMilestone) {
+    final Spec spec = TestSpecFactory.createMinimal(specMilestone);
+    final SchemaDefinitionCache cache = new SchemaDefinitionCache(spec);
+
+    final List<Method> methods = Arrays.asList(SchemaDefinitions.class.getMethods());
+    // all the get methods should return the same object on subsequent calls,
+    // as creation of the schema objects is relatively expensive
+    methods.stream()
+        .filter(method -> method.getName().startsWith("get"))
+        .forEach(
+            method -> {
+              try {
+                LOG.info(method.getName());
+                final SszSchema<?> first =
+                    (SszSchema<?>) method.invoke(cache.getSchemaDefinition(specMilestone));
+                final SszSchema<?> second =
+                    (SszSchema<?>) method.invoke(cache.getSchemaDefinition(specMilestone));
+                assertThat(first.getJsonTypeDefinition()).isEqualTo(second.getJsonTypeDefinition());
+              } catch (IllegalAccessException | InvocationTargetException e) {
+                LOG.error(e);
```

Comment:
Seems like we should fail the test if there are errors here rather than just logging (so probably `Assertions.fail(e)` I think).